app/src/main/cpp/skyline/applet/swkbd/software_keyboard_applet.h

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2022 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#pragma once
+
+#include <services/am/applet/IApplet.h>
+#include <services/applet/common_arguments.h>
+#include <jni.h>
+#include "software_keyboard_config.h"
+
+namespace skyline::applet::swkbd {
+    /**
+     * @brief Return values for swkbd
+     * @url https://switchbrew.org/wiki/Software_Keyboard#CloseResult
+     */
+    enum class CloseResult : u32 {
+        Enter = 0x0,
+        Cancel = 0x1,
+    };
+
+    /**
+     * @url https://switchbrew.org/wiki/Software_Keyboard#TextCheckResult
+     */
+    enum class TextCheckResult : u32 {
+        Success = 0x0,
+        ShowFailureDialog = 0x1,
+        ShowConfirmDialog = 0x2,
+    };
+
+    //The number of bytes used to put the text on the storages returned by swkbd
+    static constexpr u32 SwkbdTextBytes{0x7D4};
+
+    struct OutputResult {
+        CloseResult closeResult;
+        std::array<u8, SwkbdTextBytes> chars;
+
+        OutputResult(CloseResult closeResult, std::u16string_view text, bool useUtf8Storage);
+    };
+
+    struct InteractiveOutputResult {
+        u64 size;
+        std::array<u8, SwkbdTextBytes> chars;
+        u8 _pad_[0x8];
+        InteractiveOutputResult(std::u16string_view text, bool useUtf8Storage);
+    };
+
+    class SoftwareKeyboardApplet : public service::am::IApplet {
+      private:
+        std::mutex inputDataMutex;
+        std::queue<std::shared_ptr<service::am::IStorage>> normalInputData;
+        KeyboardConfigVB config{};
+        bool validation_pending{false};
```

Comment:
Use camelCase for all these variables, can drop {falce} for {} since it's implicit

Reply:
IMO it's fine keeping {false} as it expresses intention more

Reply:
We don't do elsewhere in codebase, so {} should be preferred